packages/browser-crawler/src/internals/browser-crawler.ts

```diff
@@ -802,35 +805,54 @@ interface EnqueueLinksInternalOptions {
 }
 
 /** @internal */
-export async function browserCrawlerEnqueueLinks({
-    options,
-    page,
-    requestQueue,
-    robotsTxtFile,
-    onSkippedRequest,
-    originalRequestUrl,
-    finalRequestUrl,
-}: EnqueueLinksInternalOptions) {
+interface BoundEnqueueLinksInternalOptions {
+    enqueueLinks: BasicCrawlingContext['enqueueLinks'];
+    options?: ReadonlyDeep<Omit<EnqueueLinksOptions, 'requestQueue'>> & Pick<EnqueueLinksOptions, 'requestQueue'>;
+    originalRequestUrl: string;
+    finalRequestUrl?: string;
+    page: CommonPage;
+}
+
+/** @internal */
+function isEnqueueLinksBound(
+    options: EnqueueLinksInternalOptions | BoundEnqueueLinksInternalOptions,
+): options is BoundEnqueueLinksInternalOptions {
+    return !!(options as BoundEnqueueLinksInternalOptions).enqueueLinks;
```

Comment:
```suggestion
    return Boolean((options as BoundEnqueueLinksInternalOptions).enqueueLinks);
```

I know Martin prefers the `!!` way, but perhaps I can convince you... 😁

Reply:
I still don't like those "scalar constructors", but will leave it up to you. Here I don't really see a value (I don't consider this more readable personally), but sometimes the `String(...)` constructor is needed (has different semantics from `'' + ...`)

<img width="400" height="400" alt="image" src="https://github.com/user-attachments/assets/ae593465-9e28-48fe-b1e1-2bc84b40d60a" />

Reply:
> Boolean makes you put parentheses around the thing that you're converting

Well, this is the thing I actually dislike the most about it 😅 

But again, up to you guys, this bit is not worth a lengthy discussion, I am fine with both.